### PR TITLE
No legacy

### PR DESCRIPTION
**Pull request by odolbeau**
This is a rebase of #32 
It removes a lot of classes, it will be way easier to run php-cs-fixer & PHPStan! ^^

(Need a major release)

### CODE REVIEW

**Comment by Nyholm** [comment]
Oh. Interesting. I was just it would be a BC break. I’ll review this carefully shortly.

**Comment by rvanlaak** [comment]
We can use PHPStan's baseline feature to continue, so no worries about a dependency on that 👍 

https://medium.com/@ondrejmirtes/phpstans-baseline-feature-lets-you-hold-new-code-to-a-higher-standard-e77d815a5dff

File: src/Dumper/Port/SymfonyPort.php [hunk 1]
```diff
@@ -1,102 +0,0 @@
...
-
-use Symfony\Component\Translation\MessageCatalogue;
-
-/**
- * This code is moved from the Symfony 3.4 repo. It will be removed when
- * we drop support for Symfony 3.3 and below.
- *
- * @author Tobias Nyholm <tobias.nyholm@gmail.com>
- */
-class SymfonyPort
```
Comment (Nyholm) [inline src/Dumper/Port/SymfonyPort.php:22]: Is removing this class a BC break?

File: src/FileStorage.php [hunk 2]
```diff
@@ -58,18 +58,8 @@ final class FileStorage implements Storage, TransferableStorage
      * @param array                      $dir
      * @param array                      $options
      */
-    public function __construct($writer, $reader, array $dir, array $options = [])
+    public function __construct(TranslationWriterInterface $writer, TranslationReaderInterface $reader, array $dir, array $options = [])
```
Comment (Nyholm) [inline src/FileStorage.php:61]: Removing the possibility to use `TranslationLoader` here is a BC break Im afraid.

**Review by Nyholm — APPROVED** [review]
Strictly speaking. I think there are BC breaks. 

I like that you rebased this and I do think it makes sense to let `1.1.0` (or `1.1.1`) be the last 1.x release and then release 2.0 with this change. 

It would make this library sooo much simpler. =)

**Comment by odolbeau** [comment]
I merge this PR.
Before releasing a v2 of this storage, let's wait for `php-translation/common` v2 to be released (to be done once https://github.com/php-translation/common/pull/24 is merged).

TODO before v2:
* Require `php-translation/common` v2 only
* Clean code according to #38 (+ add php-cs-fixer)

**Comment by Nyholm** [comment]
Wohoo!

**Pull request merged**